src/obb.cpp

```diff
@@ -43,6 +43,13 @@ class OBBPrivate : public FCL_OBB
 OBB::OBB()
 {
   this->obb_.reset(new OBBPrivate);
+#if FCL_MAJOR_VERSION > 0 || FCL_MINOR_VERSION > 5
+  // Zero-initialize the OBB. This is done to unify the behavior of FCL 0.5 (which zero-initializes by default) and
```

Comment:
Didn't you recently add code to explicitly initialize the OBB at the caller side (https://github.com/ros-planning/geometric_shapes/pull/231)?
This should be reverted then, shouldn't it?

Reply:
No, these two complement each other. I've tested #231 on Melodic where it worked as expected. However, on Noetic with FCL 0.6, the assumed zero-initialization does not happen. This PR adds just the zero-initialization to FCL 0.6 so that the behavior is the same for both.

It is helpful to note that zero-initialization of the OBB is different from "identity"-initialization. The FCL OBB structure contains a 3x3 matrix for rotation. If zero-initialized, it is invalid.

#231 added the "identity"-initialization so that the merging of OBBs does not start with an invalid one.

Reply:
Another option would be to always "identity"-initialize the OBB in the default constructor.

The zero-initialized OBBs were just a direct translation from FCL 0.5 where zero-initialization is the default. However, it seems to be a very bug-prone approach as invalid OBBs might quite easily get into the code.

I suggest a default-constructed OBB should be a valid one with identity rotation, zero pose and zero extents. "Uninitialized"/empty OBBs would then be represented as all OBBs with zero extents (but with valid orientation).

If this change would be made, #231 could get reverted.

Do you like this approach?

Reply:
I think that's the right way to go, particularly if you want to avoid invalid OBBs.

Reply:
Ok, I'll update the PR.

Reply:
Done. I also added a check for the "identity"-initialization by no-arg constructor.